File: src/main/java/com/hank/iterface/ReportTest.java
package com.hank.iterface;

import java.util.ArrayList;
import java.util.List;

public class ReportTest {
    public static void main(String[] args) {
        Report finance = new FinanceReport();
        Report health = new FinanceReport();
        List<Report> reports = new ArrayList<>();
        reports.add(finance);
        reports.add(health);
        for (Report s : reports){
            s.load();
            s.print();
        }
    }
}

File: src/main/kotlin/com/hank/itface/ReportTest.kt
package com.hank.itface

fun main() {
    val finance:Report = FinanceReport()
    val health:Report = HealthReport()
    val list = mutableListOf<Report>(finance,health)
    for (r:Report in list){
        r.load()
        r.print()
    }
}